versioned_docs/version-V3/guides/local-environment.mdx

```diff
@@ -4,142 +4,148 @@ title: Set Up Your Local Environment
 
-<TabItem value="yarn" label="yarn">
+Next we’ll use NPM to add the Uniswap V3 contracts which will allow us to seamlessly integrate with the protocol in our new contracts:
 
-   ```bash
-   $ yarn add @uniswap/v3-periphery
-   ```
+```bash
+$ npm add @uniswap/v3-periphery @uniswap/v3-core
+```
 
-</TabItem>
-</Tabs>
+The Uniswap V3 contracts were written using a past version of the solidity compiler. Since we’re building integrations on V3 we have to tell Hardhat to use the correct compiler to build these files. Go to the `./hardhat.config.js` file and change the Solidity version to “0.7.6”: 
 
-3. Create a new hardhat config file, which you can use for compiling and testing contracts. For an example of how a typical smart contract repo is structure, select the "create a sample project" option during setup.
+```jsx
+// ...
+module.exports = {
+  solidity: "0.7.6",
+};
+```
 
-<Tabs groupId="package-managers">
-<TabItem value="npm" label="npm">
+That’s it! You should now have a functional development environment to start building on chain Uniswap integrations. Let’s run a quick test to confirm everything is set up properly. 
+
+## Compile a Basic Contract
+
+To confirm that our environment is configured correctly we’ll attempt to compile a basic Swap contract. Create a new file, `./contracts/Swap.sol` and paste the following code into it (a detailed guide to this contract can be found [here](https://docs.uniswap.org/protocol/guides/swaps/single-swaps)): 
+
+```jsx
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import '@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol';
+import '@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol';
+
+contract SimpleSwap {
+    ISwapRouter public immutable swapRouter;
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant WETH9 = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint24 public constant feeTier = 3000;
+    
+    constructor(ISwapRouter _swapRouter) {
+        swapRouter = _swapRouter;
+    }
+    
+    function swapWETHForDAI(uint amountIn) external returns (uint256 amountOut) {
+
+        // Transfer the specified amount of WETH9 to this contract.
+        TransferHelper.safeTransferFrom(WETH9, msg.sender, address(this), amountIn);
+        // Approve the router to spend WETH9.
+        TransferHelper.safeApprove(WETH9, address(swapRouter), amountIn);
+        // Create the params that will be used to execute the swap
+        ISwapRouter.ExactInputSingleParams memory params =
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: WETH9,
+                tokenOut: DAI,
+                fee: feeTier,
+                recipient: msg.sender,
+                deadline: block.timestamp,
+                amountIn: amountIn,
+                amountOutMinimum: 0,
```

Comment:
you should generally always set _something_ here for slippage protection. I guess dont want to confuse reader but bad practice to be teaching imo

Reply:
How's this? I don't really want to add all the code to get prices. I might end up just replacing this whole example with something simpler...

Reply:
works

Reply:
usually for end users minOut is good enough, priceLimit more for advanced arbs